bin/init.sh

```diff
@@ -55,9 +55,14 @@ if [ ! -f vendor/envoy-$PROXYVERSION ] ; then
     curl -Lo - https://storage.googleapis.com/istio-build/proxy/envoy-$PROXY.tar.gz | tar xz
     cp usr/local/bin/envoy $ISTIO_GO/vendor/envoy-$PROXYVERSION
```

Comment:
When I saw the proxy PR that removed this I didn't see an explanation for this part of the change that you're essentially reverting.  Neither way looked quite right to me.  I think what's needed here is the revert plus adding a:

rm -f $GO_TOP/bin/envoy

after the extraction of the envoy tar.  This will force the $GO_TOP/bin/envoy to get updated when the proxy version changes.

Reply:
Moved this to separate PR, will merge it. Yes, it needs to remove the old one, added some comments as well in https://github.com/istio/istio/pull/2654